This patch is a backport from binutils-gdb commit:
a3972330f4: Fix a missing include of <string>

This code is free software; you can redistribute it and/or modify it
under the terms of the GNU General Public License as published by the
Free Software Foundation; either version 3, or (at your option) any
later version.

This code is distributed in the hope that it will be useful, but WITHOUT
ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

diff -Nur binutils-gdb/gold/errors.h binutils-gdb-oracle/gold/errors.h
--- binutils-gdb/gold/errors.h	2022-06-16 17:49:08.412691685 -0500
+++ binutils-gdb-oracle/gold/errors.h	2022-06-16 18:34:19.463006903 -0500
@@ -24,6 +24,7 @@
 #define GOLD_ERRORS_H
 
 #include <cstdarg>
+#include <string>
 
 #include "gold-threads.h"
 
